gprofiler/spark/sampler.py

```diff
@@ -93,16 +93,18 @@ def _get_yarn_config_property(
                         return value_property.text
         return default
 
-    def _guess_driver_application_master_address(self, process: psutil.Process) -> str:
+    def _guess_standalone_master_webapp_address(self, process: psutil.Process) -> str:
         """
-        Selects the master address for a org.apache.spark.deploy.master.Master running on this node.
-        Uses master_address if given, or defaults to my hostname.
+        Selects the master address for a standalone cluster.
+        Uses master_address if given.
         """
-        if self._master_address is not None:
+        if self._master_address:
             return self._master_address
         else:
-            host_name = get_hostname()
-            return host_name + ":4040"
+            master_process_args = process.cmdline()
+            master_ip = master_process_args[master_process_args.index("--host") + 1]
```

Comment:
Consider extracting the pattern `x[x.index(y) + 1]` to an appropriately named helper function, so it's clear what the result is going to be.

Reply:
https://github.com/Granulate/gprofiler/pull/718/commits/ca6a5d743f86f062822d249bf15fb2f47f0974f9